File: client/GhostClient/Core/Network/Packets/MatchStatusPacket.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using GhostClient.Core;

namespace Ghost.Core.Network.Packets
{
    public class MatchStatusPacket : Packet
    {
        public MatchStatusPacket(Stream stream) : base(stream)
        {
        }

        public MatchStatusPacket(Stream stream, bool autoClose) : base(stream, autoClose)
        {
        }

        public MatchStatusPacket(byte[] data) : base(data)
        {
        }

        protected override void OnHandlePacket()
        {
            bool state = Consume(1).BoolValue;
            int reasonLength = Consume(4).IntValue;
            string reason = Consume(reasonLength).AsciiStringValue;

            GameHandler.Game.UpdateStatus(state, reason);
        }
    }
}

File: client/core/src/me/eddiep/ghost/client/network/packets/MatchStatusPacket.kt
package me.eddiep.ghost.client.network.packets

import me.eddiep.ghost.client.network.Packet
import me.eddiep.ghost.client.network.PlayerClient

class MatchStatusPacket : Packet<PlayerClient>() {

    override fun handle() {
        val state = consume(1).asBoolean()
        val reasonLength = consume(4).asInt()
        val reason = consume(reasonLength).asString()

        client.game.updateStatus(state, reason)
    }
}
